conda_lock.py

```diff
@@ -17,19 +17,28 @@
 import yaml
 import os
 import requests
+import tempfile
+
 
 DEFAULT_PLATFORMS = ["osx-64", "linux-64", "win-64"]
-FAKE_PREFIX_NAME = "__magicmarker"
-FAKE_PKGS_ROOT = "/tmp/something/that/does/not/exist"
 
 
+def _safe_fake_prefix_name():
+    args = ("conda", "env", "list", "--quiet", "--json")
+    json_output = subprocess.check_output(args)
+    envs = json.loads(json_output)["envs"]
+    PREFIX_NAMES = [pathlib.Path(env).stem for env in envs if "envs" in env]
+
+    while (FAKE_PREFIX_NAME := next(tempfile._get_candidate_names())) not in PREFIX_NAMES:
```

Comment:
I don't like this b/c it will limit to Pyhton 3.8 and uses a private method.

Reply:
basically we just need a way to get a thing to find a writable prefix (which is somehow required for conda to solve even in dry-run mode)

Reply:
So we can just make a new tempdir and pick a random prefix inside there